packages/suite/src/reducers/wallet/coinjoinReducer.ts

```diff
@@ -241,6 +245,71 @@ const updateClientStatus = (
     };
 };
 
+const appendSessionPhase = (
+    draft: CoinjoinState,
+    payload: ExtractActionPayload<typeof COINJOIN.CLIENT_SESSION_PHASE>,
+) => {
+    const accounts = payload.accountKeys.flatMap(
+        accountKey => draft.accounts.find(({ key }) => key === accountKey) || [],
+    );
+
+    if (!accounts || !accounts.length) {
+        return;
+    }
+
+    accounts.forEach(({ session }) => {
+        if (!session) {
+            return;
+        }
+
+        const isPreviousRoundPhase =
+            getRoundPhaseFromSessionPhase(payload.phase) <
+            getRoundPhaseFromSessionPhase(session?.sessionPhaseQueue.at(-1) || 0);
+        const isSameSessionPhase = payload.phase === session?.sessionPhaseQueue.at(-1);
+
+        const isFirstPhase =
+            getRoundPhaseFromSessionPhase(payload.phase) === RoundPhase.InputRegistration;
+
+        if (isSameSessionPhase || (isPreviousRoundPhase && !isFirstPhase)) {
+            return;
+        }
+
+        const timeSinceLastChange =
+            Date.now() - (session.lastSessionPhaseChangeTimestamp || Date.now());
+
+        if (timeSinceLastChange > SESSION_PHASE_TRANSITION_DELAY) {
+            session.sessionPhaseQueue = [payload.phase];
```

Comment:
doesn't this potentially skip some phases?

Reply:
How do you think that can happen?

Reply:
756ffcfe8e31bac98c7b3a57b803f565d5a32286 4ca873c7da559912600c9f863fcd8778877235d4

Reply:
I believe that it works as expected now.